strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/ProxyRepositoryProvider.java

```diff
@@ -87,14 +93,21 @@ public RepositoryPath resolvePathForceFetch(RepositoryPath repositoryPath)
             throw e;
         }
     }
-    
+
+    private RepositoryPath handleRepositoryPathExpiration(final RepositoryPath repositoryPath)
+    {
+        final LayoutProvider layoutProvider = layoutProviderRegistry.getProvider(
+                repositoryPath.getRepository().getLayout());
+        return layoutProvider.handleRepositoryPathExpiration(repositoryPath);
```

Comment:
I think this probably better to be done with sync event instead of explicit new method in `LayoutProvider`.
You can have a look at `ProxyRepositoryProvider.search` method which handle similar case with search against metadata in remote repositories.

Reply:
I like this idea, thanks 👍